tests/test_web/test_broker_error.py

```diff
@@ -0,0 +1,218 @@
+from unittest import mock
+
+from kombu.exceptions import OperationalError
+
+from iib.web.models import Request, RequestStateMapping, RequestState, RequestAdd, RequestRm
+
+
+def assert_testing(rv, mock_smfsc, db):
+    response_json = {'error': 'The scheduling of the build request with ID 1 failed'}
+    assert rv.status_code == 500
+    assert response_json == rv.json
+    assert mock_smfsc.call_count == 2
+
+    req_state = db.session.query(Request).filter(Request.id == 1).one()
+    assert req_state.state.state == RequestStateMapping.failed.value
+
+
+@mock.patch('iib.web.api_v1.handle_add_request')
+@mock.patch('iib.web.api_v1.messaging.send_message_for_state_change')
+def test_catch_add_bundle_failure(mock_smfsc, mock_har, db, auth_env, client):
+    mock_har.apply_async.side_effect = OperationalError
+    data = {
+        'bundles': ['some:thing'],
+        'binary_image': 'binary:image',
+        'add_arches': ['s390x'],
+        'organization': 'org',
+        'cnr_token': 'token',
+        'overwrite_from_index': True,
+    }
+
+    rv = client.post('/api/v1/builds/add', json=data, environ_base=auth_env)
+    mock_har.apply_async.assert_called_once()
+    assert_testing(rv, mock_smfsc, db)
+
+
+@mock.patch('iib.web.api_v1.handle_regenerate_bundle_request')
+@mock.patch('iib.web.api_v1.messaging.send_message_for_state_change')
+def test_catch_regenerate_bundle_failure(mock_smfsc, mock_hrbr, db, auth_env, client):
+    mock_hrbr.apply_async.side_effect = OperationalError
+
+    data = {
+        'from_bundle_image': 'registry.example.com/bundle-image:latest',
+    }
+
+    rv = client.post('/api/v1/builds/regenerate-bundle', json=data, environ_base=auth_env)
+    mock_hrbr.apply_async.assert_called_once()
+    assert_testing(rv, mock_smfsc, db)
+
+
+@mock.patch('iib.web.api_v1.handle_rm_request')
+@mock.patch('iib.web.api_v1.messaging.send_message_for_state_change')
+def test_catch_remove_operator_failure(mock_smfsc, mock_rm, db, auth_env, client):
+    mock_rm.apply_async.side_effect = OperationalError
+    data = {
+        'operators': ['some:thing'],
+        'binary_image': 'binary:image',
+        'from_index': 'index:image',
+    }
+
+    rv = client.post('/api/v1/builds/rm', json=data, environ_base=auth_env)
+    mock_rm.apply_async.assert_called_once()
+    assert_testing(rv, mock_smfsc, db)
+
+
+@mock.patch('iib.web.api_v1.handle_regenerate_bundle_request')
+@mock.patch('iib.web.api_v1.messaging.send_message_for_state_change')
+@mock.patch('iib.web.api_v1.messaging.send_messages_for_new_batch_of_requests')
+def test_catch_regenerate_bundle_batch_failure(
+    mock_smfnbor, mock_smfsc, mock_hrbr, app, auth_env, client, db,
+):
+
+    user_to_queue = {'tbrady@DOMAIN.LOCAL': 'Patriots'}
+    expected_queue = 'Patriots'
+    annotations = None
+
+    mock_hrbr.apply_async.side_effect = OperationalError
+    app.config['IIB_USER_TO_QUEUE'] = user_to_queue
```

Comment:
This is all sufficiently tested elsewhere. I'd like to avoid duplicating tests. Could you please remove this part of the test?